internal/bzlmod/go_deps.bzl

```diff
@@ -433,6 +450,19 @@ _gazelle_override_tag = tag_class(
             extension within this Bazel module.""",
             mandatory = True,
         ),
+        "build_file_generation": attr.string(
+            default = "auto",
+            doc = """One of `"auto"` (default), `"on"`, `"off"`.
+
+            Whether Gazelle should generate build files for the Go module. In
```

Comment:
I was going to suggest capitalizing "build" as "BUILD" here and on the next line, but I see that [the existing documentation for `go_repository`'s "build_file_generation" attribute](https://github.com/bazelbuild/bazel-gazelle/blob/master/repository.md#user-content-go_repository-build_file_generation) writes it in lowercase too.